### ISSUE TITLE
Select widget in admin does not render correctly

### ISSUE THREAD

**Issue by patta42** [wagtail/wagtail#4063]
### Issue Summary

The CSS for the admin interface requires a `<select>` to be within an `<li>` with class `typed_choice_field` to render the select box with a dropdown-indicator (the caret on the right of the field). If one overrides the widget on a `CharField` that does not have a `choices` property, the surrounding `<li>` does not receive the class `typed_choice_field` and thus the `<select>` renders without the dropdown indicator.

### Steps to Reproduce

For example, use the following snippet somewhere in your project:

```

from django.forms.widgets import Select

# add usual imports for a model.py file

@register_snippet
class Test( models.Model ):
    good = models.CharField(
        max_length = 1,
        choices = (('a','a'),('b','b'))
    )
    bad = models.CharField(
        max_length = 1,
    )

    panels = [
        FieldPanel('good'),
        FieldPanel(
            'bad',
            widget = Select(
                choices = (
                    ('a','a'), ('b','b')
                )
            )
        )
    ]

```
Now look at the form in the admin for the snippet. You will see that the form-field for `bad` looks like a normal `<input type="text">`, although it behaves live a `<select>`.

I think that a styled `<select>` should always resemble the look of a non-styled one, thus it should have a dropdown indicator, no matter what the parent elements are.

### Technical details

* Python version: 3.4.6.
* Django version: 1.11.7
* Wagtail version: 1.13.1
* Browser version: Chrome 62, Linux

**Comment by enzedonline** [comment]
Old issue but still exists. I'm coming across this because I need to set the choices dynamically at the panels declaration.
Appearance is a text field with the first option selected as default.
I needed to add a dummy blank entry to the start of the dynamic list, but the appearance is quite misleading for the content editor. 
Would be good to see this fixed.

**Comment by enzedonline** [comment]
Update - I got around this by subclassing the FieldPanel and overriding the field_type function to force it to the right type:
```
    def field_type(self):
        return 'typed_choice_field' 
```

**Comment by thibaudcolas** [comment]
This has been fixed as of #8921. Now the select styles only depend on `<select>` element, and all `<select>` elements across the admin should render the same.

Note there are still a few fields where the _layout_ of elements around the widget will differ based on the field type, but not `select`.
